storage/innobase/ibuf/ibuf0ibuf.cc

```diff
@@ -3812,7 +3813,15 @@ static bool ibuf_restore_pos(
     return true;
   }
 
-  if (fil_space_get_flags(space) == UINT32_UNDEFINED) {
+  /* Check if the tablespace is dropped. */
+  fil_space_t *sp = fil_space_acquire_silent(space);
+  raii::Sentry<> space_release_guard{[&]() -> void {
+    if (sp != nullptr) {
+      fil_space_release(sp);
+    }
+  }};
+
+  if (sp == nullptr || sp->is_deleted() || sp->flags == UINT32_UNDEFINED) {
```

Comment:
Hmm... I have doubts about such fil_space_t::is_deleted() usage - I think it requires caller to own shard mutex...
`
bool fil_space_t::is_deleted() const {
  ut_ad(fil_system->shard_by_id(id)->mutex_owned());
  return m_deleted;
}
`

Reply:
Good catch. I think I need a fil_sys wrapper

Reply:
A little deeper analysis on this:
1. in fil_close_tablespace() we first check pending operations. This uses the flag stop_new_ops to true (the same flag used by fil_space_acquire() which makes it return nullptr )
2. then space->set_deleted()
3. also adds to m_deleted_spaces (right, innodb has two of them to track deleted talbespaces :( )

Since we do a fil_space_acuire() which takes the shard mutex and checks stop_new_ops flag, it is sufficient.

Reply:
we might remove space->flags check too but leaving it for legacy reasons